### ISSUE TITLE
apache-airflow[google_auth] is not working (2.0.1)

### ISSUE THREAD

**Issue by klalafaryan** [apache/airflow#15508]
**Apache Airflow version**: 2.0.1

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`): 1.19.6"

**Environment**:

- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:  When trying to enable `apache-airflow[google_auth]`,  I am getting following exception, althought the `google_auth` dependencies have been installed.

```
kubectl logs -f pods/airflow-web-7d675f8c6f-lwhjx -n airflow airflow-web
/opt/python/site-packages/airflow/configuration.py:332 DeprecationWarning: The logging_level option in [core] has been moved to the logging_level option in [logging] - the old setting has been used, but please update your config.
/opt/python/site-packages/airflow/configuration.py:332 DeprecationWarning: The remote_logging option in [core] has been moved to the remote_logging option in [logging] - the old setting has been used, but please update your config.
/opt/python/site-packages/airflow/configuration.py:332 DeprecationWarning: The remote_base_log_folder option in [core] has been moved to the remote_base_log_folder option in [logging] - the old setting has been used, but please update your config.
/opt/python/site-packages/sqlalchemy/orm/relationships.py:3441 SAWarning: relationship 'DagRun.serialized_dag' will copy column serialized_dag.dag_id to column dag_run.dag_id, which conflicts with relationship(s): 'TaskInstance.dag_run' (copies task_instance.dag_id to dag_run.dag_id), 'DagRun.task_instances' (copies task_instance.dag_id to dag_run.dag_id). If this is not the intention, consider if these relationships should be linked with back_populates, or if viewonly=True should be applied to one or more if they are read-only. For the less common case that foreign key constraints are partially overlapping, the orm.foreign() annotation can be used to isolate the columns that should be written towards.   The 'overlaps' parameter may be used to remove this warning. (Background on this error at: http://sqlalche.me/e/14/qzyx)
/opt/python/site-packages/sqlalchemy/orm/relationships.py:3441 SAWarning: relationship 'SerializedDagModel.dag_runs' will copy column serialized_dag.dag_id to column dag_run.dag_id, which conflicts with relationship(s): 'TaskInstance.dag_run' (copies task_instance.dag_id to dag_run.dag_id), 'DagRun.task_instances' (copies task_instance.dag_id to dag_run.dag_id). If this is not the intention, consider if these relationships should be linked with back_populates, or if viewonly=True should be applied to one or more if they are read-only. For the less common case that foreign key constraints are partially overlapping, the orm.foreign() annotation can be used to isolate the columns that should be written towards.   The 'overlaps' parameter may be used to remove this warning. (Background on this error at: http://sqlalche.me/e/14/qzyx)
/home/airflow/.local/lib/python3.8/site-packages/azure/cosmos/session.py:186 SyntaxWarning: "is not" with a literal. Did you mean "!="?
[2021-04-22 12:46:09,330] {providers_manager.py:295} WARNING - Exception when importing 'airflow.providers.microsoft.azure.hooks.wasb.WasbHook' from 'apache-airflow-providers-microsoft-azure' package: No module named 'azure.storage.blob'
[2021-04-22 12:46:09,982] {providers_manager.py:295} WARNING - Exception when importing 'airflow.providers.microsoft.azure.hooks.wasb.WasbHook' from 'apache-airflow-providers-microsoft-azure' package: No module named 'azure.storage.blob'
  ____________       _____________
 ____    |__( )_________  __/__  /________      __
____  /| |_  /__  ___/_  /_ __  /_  __ \_ | /| / /
___  ___ |  / _  /   _  __/ _  / / /_/ /_ |/ |/ /
 _/_/  |_/_/  /_/    /_/    /_/  \____/____/|__/
[2021-04-22 12:46:10,230] {dagbag.py:448} INFO - Filling up the DagBag from /dev/null
Traceback (most recent call last):
  File "/home/airflow/.local/bin/airflow", line 8, in <module>
    sys.exit(main())
  File "/opt/python/site-packages/airflow/__main__.py", line 40, in main
    args.func(args)
  File "/opt/python/site-packages/airflow/cli/cli_parser.py", line 48, in command
    return func(*args, **kwargs)
  File "/opt/python/site-packages/airflow/utils/cli.py", line 89, in wrapper
    return f(*args, **kwargs)
  File "/opt/python/site-packages/airflow/cli/commands/webserver_command.py", line 360, in webserver
    app = cached_app(None)
  File "/opt/python/site-packages/airflow/www/app.py", line 135, in cached_app
    app = create_app(config=config, testing=testing)
  File "/opt/python/site-packages/airflow/www/app.py", line 113, in create_app
    init_appbuilder(flask_app)
  File "/opt/python/site-packages/airflow/www/extensions/init_appbuilder.py", line 46, in init_appbuilder
    AirflowAppBuilder(
  File "/opt/python/site-packages/flask_appbuilder/base.py", line 148, in __init__
    self.init_app(app, session)
  File "/opt/python/site-packages/flask_appbuilder/base.py", line 202, in init_app
    self.sm = self.security_manager_class(self)
  File "/opt/python/site-packages/airflow/www/security.py", line 160, in __init__
    super().__init__(appbuilder)
  File "/opt/python/site-packages/flask_appbuilder/security/sqla/manager.py", line 51, in __init__
    super(SecurityManager, self).__init__(appbuilder)
  File "/opt/python/site-packages/flask_appbuilder/security/manager.py", line 250, in __init__
    from authlib.integrations.flask_client import OAuth
ModuleNotFoundError: No module named 'authlib'
```

**What you expected to happen**:

Airflow web container should start without any errors.

**How to reproduce it**:

Run the airflow container (`apache/airflow:2.0.1-python3.8`) on kubernetes with `apache-airflow[google_auth]` dependency.

**Anything else we need to know**:

I am using the [helm chart](https://github.com/airflow-helm/charts) for Airflow, and I thought that the problem is coming from the chart itselft, but in the end I have found out that the problem is coming from `apache-airflow[google_auth]`

You can find [here](https://github.com/airflow-helm/charts/issues/178) the helm chart ticket.

**Comment by potiuk** [comment]
The way how extra dependencies are installed in the helm chart is causing this. There is nothing we can do in airflow.

The problem is that specifying `extraPipPackages' in the helm chart, make it reinstall a new version of Airlfow in a different site location and it causes that the module cannot be imported. There is no way Airflow community can do anything about it. 

I suggested the way to fix, commenting on @thesuperzapper proposal here: https://github.com/airflow-helm/charts/issues/169#issuecomment-824955814

**Comment by thesuperzapper** [comment]
For reference, this has been fixed in the `8.0.9` version of the [stable/airflow](https://github.com/airflow-helm/charts/tree/main/charts/airflow) helm chart.
